Forte.ContentfulSchema/Core/ContentTypeDefinition.cs

```diff
@@ -80,6 +80,18 @@ public bool Update(ContentType contentType)
                     this.UpdateField(match.Existing, match.Updated, ref modified); 
                 }
             }
+
+            if (InferedContentType.Fields.Count == contentType.Fields.Count)
+            {
+                for (var i = 0; i < InferedContentType.Fields.Count; i++)
+                {
+                    if (InferedContentType.Fields[i]?.Id != contentType.Fields[i]?.Id) // order has changed
```

Comment:
@mgrzyb I feel like this is definitely enough to check the order. I've added to test cases and it works properly.

Whole solution still works:
![image](https://user-images.githubusercontent.com/14861637/85201395-38594300-b2ff-11ea-971f-e3827cc25814.png)